tests/conftest.py

```diff
@@ -34,16 +35,61 @@
 pytest_plugins = ['aiida.tools.pytest_fixtures', 'sphinx.testing.fixtures']
 
 
+def pytest_collection_modifyitems(items):
+    """Automatically generate markers for certain tests.
+
+    Most notably, we add the 'presto' marker for all tests that
+    are not marked with either requires_rmq or requires_pgsql.
+    """
+    filepath_psqldos = Path(__file__).parent / 'storage' / 'psql_dos'
+    filepath_django = Path(__file__).parent / 'storage' / 'psql_dos' / 'migrations' / 'django_branch'
+    filepath_sqla = Path(__file__).parent / 'storage' / 'psql_dos' / 'migrations' / 'sqlalchemy_branch'
+
+    for item in items:
+        filepath_item = Path(item.fspath)
+
+        # Add 'nightly' marker to all tests in 'storage/psql_dos/migrations/<django|sqlalchemy>_branch'
+        if filepath_item.is_relative_to(filepath_django) or filepath_item.is_relative_to(filepath_sqla):
+            item.add_marker('nightly')
+
+        # Add 'requires_rmq' for all tests that depend on one or more *daemon* fixtures
+        if 'daemon' in ''.join(item.fixturenames):
+            item.add_marker('requires_rmq')
```

Comment:
This feels very fragile. We could add a fixture in the future that contains `daemon` but doesn't actually require RMQ. Think it is better to be conservative and explicitly match the fixtures we have know that we know use RMQ.

Reply:
Agreed. Turns out we can simply match against the `daemon_client` fixture on which all other depend on (`item.fixturename` contains even transitive fixtures for a given test)